drupal/web/modules/custom/covid/src/Normalizer.php

```diff
@@ -1,27 +1,45 @@
 <?php
 
-
 namespace Drupal\covid;
 
-
+use Drupal\Core\Field\Plugin\Field\FieldType\StringLongItem;
 use Drupal\Core\TypedData\Plugin\DataType\StringData;
 use Drupal\link\Plugin\Field\FieldType\LinkItem;
 use Drupal\serialization\Normalizer\TypedDataNormalizer;
-
+use Drupal\text\Plugin\Field\FieldType\TextLongItem;
+use Drupal\text\TextProcessed;
+
+/**
+ * Class Normalizer.
+ *
+ * @package Drupal\covid
+ */
 class Normalizer extends TypedDataNormalizer {
 
+  /**
+   * {@inheritdoc}
+   */
   public function normalize($object, $format = NULL, array $context = []) {
     $data = parent::normalize($object, $format, $context);
-
-    if ($object instanceof StringData && ($parent = $object->getParent()) instanceof LinkItem && $object->getString() === "") {
-      $parsedUrl = parse_url($parent->get('uri')->getValue());
-      $url = '';
-      foreach (['host', 'path'] as $part) {
-        if (isset($parsedUrl[$part])) {
-          $url .= $parsedUrl[$part];
+    $parent = $object->getParent();
+    if ( in_array(get_class($object), [StringData::class, StringLongItem::class])) {
```

Comment:
Tohle mi přijde hrozně zmatečné. Nemůžeme radši řešit ve dvou úplně oddělených větvích?

Reply:
kvoli comu? jedna podmienka by bola iba na string data ale druha by stale zostala s oboma classami.